Source/WebCore/platform/graphics/cocoa/VideoMediaSampleRenderer.mm

```diff
@@ -47,97 +48,226 @@ - (void)resetUpcomingSampleBufferPresentationTimeExpectations;
 namespace WebCore {
 
 VideoMediaSampleRenderer::VideoMediaSampleRenderer(WebSampleBufferVideoRendering *renderer)
-    : m_renderer(renderer)
+    : m_workQueue(WorkQueue::create("VideoMediaSampleRenderer Queue"_s))
+    , m_renderer(renderer)
 {
 }
 
 VideoMediaSampleRenderer::~VideoMediaSampleRenderer()
 {
-    [m_renderer flush];
-    [m_renderer stopRequestingMediaData];
-    if (m_decompressionSession)
-        m_decompressionSession->invalidate();
+    m_workQueue->dispatchSync([&] {
```

Comment:
rather than a dispatchSync we could do `m_workQueue->dispatch([decompressionSession = std::exchange(m_decompressionSession, { })] { decompressionSession->invalidata(); });`